### PR TITLE
Replace folds with sum()

### PR DESCRIPTION
**Pull request by urschrei**
Removes some of the unnecessary `fold`s (see #254). WIP til I've tracked them all down and checked to see whether the lifetime issues that crop up with the remaining uses can be resolved.

### CODE REVIEW

**Comment by urschrei** [comment]
I think this actually covers all the cases in which it's just a simple accumulating fold.

File: geo/src/algorithm/euclidean_length.rs [hunk 1]
```diff
@@ -35,18 +36,18 @@ where
...
         self.lines()
             .map(|line| line.euclidean_length())
-            .fold(T::zero(), |total_length, length| total_length + length)
+            .sum()
     }
 }
 
 impl<T> EuclideanLength<T> for MultiLineString<T>
 where
-    T: Float,
```
Comment (frewsxcv) [inline geo/src/algorithm/euclidean_length.rs]: Did you mean to convert the `fold` usage below into `sum`?

File: geo/src/algorithm/centroid.rs [hunk 2]
```diff
@@ -157,7 +161,7 @@ where
 
 impl<T> Centroid<T> for MultiPolygon<T>
 where
-    T: Float + FromPrimitive,
+    T: Float + FromPrimitive + Sum,
```
[the remark this answers is not in the document]
Comment (urschrei) [reply geo/src/algorithm/centroid.rs]: Oops, removing…

File: geo/src/algorithm/euclidean_length.rs (hunk 1 above)
Comment (urschrei) [reply geo/src/algorithm/euclidean_length.rs]: It's needed because of the `Sum` bound on `EuclideanLength` for `LineString`.

**Pull request merged**